src/librustc/hir/mod.rs

```diff
@@ -1905,19 +1905,27 @@ pub struct PolyTraitRef {
 
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub enum Visibility {
-    Public,
-    Crate(CrateSugar),
-    Restricted { path: P<Path>, id: NodeId },
+    Public(Span),
+    Crate(Span, CrateSugar),
+    Restricted { span: Span, path: P<Path>, id: NodeId },
```

Comment:
Could you mirror what https://github.com/rust-lang/rust/pull/47799 did with `ast::Visibility` instead?